File: leet/ac/1207.unique-number-of-occurrences.cpp
class Solution {
  public:
    bool uniqueOccurrences(vector<int> &arr) {
        std::ios::sync_with_stdio(false), cin.tie(NULL), cout.tie(NULL);
        unordered_map<int, int> ca, ac;
        for (auto i : arr) ac[i + 1000]++;

        for (auto &i : ac) {
            ca[i.second]++;
            if (ca[i.second] > 1) return false;
        }

        return true;
    }
};

File: leet/do/1207.unique-number-of-occurrences.swift
class Solution {
    func uniqueOccurrences(_ arr: [Int]) -> Bool {
		var ca = [Int : Int]()
		var ac = [Int : Int]()

		for i in arr {
			if ac[i] == nil {
				ac[i] = 0
			}
			ac[i] += 1
		}

		for (k, v) in ac {
			ca[v] += 1
			if ca[v] > 1 {
				return false
			}
		}

		return true
    }
}
